app/controllers/times_controller.rb

```diff
@@ -25,22 +25,36 @@ def index
   end
 
   def update_duration_from_params
-    hour = params[:timeslot_hour].to_i
-    minute = params[:timeslot_minute].to_i
-    @duration = hour * 60 + minute
     @project.update(duration: @duration)
   end
 
-  def no_project_date_time?(date_time, is_date)
-    @project.project_times.where(date_time: DateTime.parse(date_time), is_date: is_date).blank?
+  def validate_times(date, is_date)
+    datetime = DateTime.parse(date)
```

Comment:
Nice job with the validating times part! Makes adding times a lot cleaner and less buggy.